.github/workflows/create-dev-tag.yaml

```diff
@@ -91,6 +91,20 @@ jobs:
 
       - name: Create tag
         run: |
+          # remove prefix and suffix alphabets
+          TAG_VERSION=$(echo "$TAG" | sed 's/^v//;s/-.*//')
+
+          MAJOR=$(echo "$TAG_VERSION" | cut -d'.' -f1)
+          MINOR=$(echo "$TAG_VERSION" | cut -d'.' -f2)
+
+          echo "Major Version: $MAJOR"
+          echo "Minor Version: $MINOR"
+
+          if [ "$MAJOR" -gt 2 ] || { [ "$MAJOR" -eq 2 ] && [ "$MINOR" -ge 15 ]; }; then
+            echo "It's depricated for v2.15 and above"
+            exit 1
```

Comment:
will this make the GHA run fail?

Reply:
You can drop this entire snippet and add a `if` condition at the job. 

After the `generate-matrix` job is run, it emits an output which looks like
```
["release-2.12","release-2.13","release-2.14","main","release-2.15","release-2.8"]
```
The `create-dev-tag` job is run for each element in this array.

We can add a `if` conditional to check `branch` is either `main` or `release-2.15` (i dunno why we need that release branch anymore but its fine ig) and then run the job only if branch is neither of those. It will "skip" the job instead of failing the job as this PR does.